runtime/rasdump/TextFileStream.cpp

```diff
@@ -73,7 +74,9 @@ TextFileStream::close(void)
 			j9file_write_text(_FileHandle, _Buffer, _BufferPos);
 		}
 		j9file_sync(_FileHandle);
-		j9file_close(_FileHandle);
+		if (_FileHandle != J9PORT_TTY_OUT && _FileHandle != J9PORT_TTY_ERR) {
```

Comment:
This condition should be base on a local variable which is set only after a successful call to `j9file_open` and not make assumptions about the values that function returns.

Reply:
I was thinking :
`if ((_FileHandle != J9PORT_TTY_OUT) && (_FileHandle != J9PORT_TTY_ERR) && _FileHandle != -1)`
rather than adding a local variable. Is there a more elegant solution?

Reply:
A local variable is not enough; we need a field to capture that state. I don't know about 'elegant', but it's definitely cleaner than making assumptions about `j9file_open` return values.

Reply:
It seems like the only time where `j9file_open` is not open is when `_FileHandle` = -1. Can't we just simply do a check for `_FileHandle == -1` ? That check would simply be isOpen().

I might be missing something here... Please let me know if that's the case

Reply:
I don't really follow either. The code is setting _FileHandle to J9PORT_TTY_OUT and J9PORT_TTY_ERR and not calling j9file_open(). If j9file_open() was called and somehow returned J9PORT_TTY_OUT or J9PORT_TTY_ERR, wouldn't closing these be wrong?

Reply:
Yes, if j9file_open() returned J9PORT_TTY_OUT or J9PORT_TTY_ERR, then _not_ closing it would be wrong.

This can really happen. Take a Linux system for example, and imagine file descriptor 2 was closed. The next call to j9file_open() might return 2 which should be closed, but the code as currently written would not.

Reply:
> Yes, if j9file_open() returned J9PORT_TTY_OUT or J9PORT_TTY_ERR, then _not_ closing it would be wrong.

I'm confused. If that's the case then when do we ever close it?

Reply:
https://github.com/eclipse/openj9/pull/4521#discussion_r254310704

@keithc-ca Didn't you say in this comment that we should not close the file stream if we are printing to STDOUT/STDERR. 


> Yes, if j9file_open() returned J9PORT_TTY_OUT or J9PORT_TTY_ERR, then _not_ closing it would be wrong.
> 
> This can really happen. Take a Linux system for example, and imagine file descriptor 2 was closed. The next call to j9file_open() might return 2 which should be closed, but the code as currently written would not.

Isn't file descriptor 2 the same as returning J9_TTY_ERR, and thus, the file stream was not even open to begin with?